libexec/rtld-elf/rtld_c18n.c

```diff
@@ -861,18 +861,18 @@ tramp_pgs_append(const struct tramp_data *data)
 	}
 	assert(cheri_gettag(tramp));
 
-	bufp = tramp + (bufp - buf);
 	memcpy(tramp, buf, len);
 
+	header = (struct tramp_header *)(tramp + (bufp - buf));
```

Comment:
Thanks, this is indeed clearer.  I would maybe put this line under the comment block for style reasons (style(9) prefers a blank line before comments), or alternatively move the blank line above after this.